wpilibNewCommands/src/main/java/edu/wpi/first/wpilibj2/command/button/Trigger.java

```diff
@@ -24,6 +24,34 @@
  * <p>This class is provided by the NewCommands VendorDep
  */
 public class Trigger implements BooleanSupplier {
+  /**
+   * Enum specifying the initial state to use for a binding. This impacts whether or not the binding
+   * will be triggered immediately.
+   */
+  public enum InitialState {
+    /**
+     * Indicates the binding should use false as the initial value. This causes a rising edge at the
+     * start if and only if the condition starts true.
+     */
+    FALSE,
+    /**
+     * Indicates the binding should use true as the initial value. This causes a falling edge at the
+     * start if and only if the condition starts false.
+     */
+    TRUE,
+    /**
+     * Indicates the binding should use the trigger's condition as the initial value. This never
+     * causes an edge at the start.
+     */
+    CONDITION,
```

Comment:
I agree that we should document that `CONDITION` is the default in the enum, but I'm not completely sure how much needs to be added to the method docs. Java has two different overloads (and the overload with the parameter defaulted mentions the behavior in the javadoc) and C++ uses a defaulted parameter, which shows up in the generated docs.
<img width="554" alt="image" src="https://github.com/user-attachments/assets/19c5528d-6a81-46b6-85c9-e31cdf0dd0b8">

Reply:
Ended up adding it anyways, since now in Java it's a little more obvious what the default behavior is and in C++ users don't need to look up what `InitialState::kConstant` means when they see it in the docs.